src/components/side_panel/data_validation/dv_editor/dv_editor.ts

```diff
@@ -134,6 +144,24 @@ export class DataValidationEditor extends Component<Props, SpreadsheetChildEnv>
   }
 
   get errorMessages(): string[] {
-    return this.state.errors.map((error) => DVTerms.Errors[error] || DVTerms.Errors.Unexpected);
+    return this.state.errors.map((error) => this.errorMessage(error));
+  }
+
+  errorMessage(reason: CancelledReason): string {
+    switch (reason) {
+      case CommandResult.TargetOutOfSheet:
+      case CommandResult.InvalidRange:
+        return DVTerms.Errors[reason](this.firstInvalidRangeString);
+      default:
+        return DVTerms.Errors[reason]() || DVTerms.Errors.Unexpected();
+    }
+  }
+
+  get firstInvalidRangeString(): string | undefined {
```

Comment:
why stop at the first ? it doesn't cost anything to return them all and join them in the error string no ?